pkg/controllers/externalsecret/externalsecret_controller.go

```diff
@@ -433,34 +433,47 @@ func hashMeta(m metav1.ObjectMeta) string {
 	})
 }
 
-func shouldSkipClusterSecretStore(r *Reconciler, es esv1beta1.ExternalSecret) bool {
+func shouldSkipClusterSecretStore(r *Reconciler, es *esv1beta1.ExternalSecret) bool {
 	return !r.ClusterSecretStoreEnabled && es.Spec.SecretStoreRef.Kind == esv1beta1.ClusterSecretStoreKind
 }
```

Comment:
If we are using pointers to this function now, we should check that `es` itself is not null before using it